pandas/_libs/window.pyx

```diff
@@ -443,10 +444,14 @@ def roll_sum(ndarray[double_t] input, int64_t win, int64_t minp,
         double val, prev_x, sum_x = 0
         int64_t s, e
         int64_t nobs = 0, i, j, N
-        bint is_variable
+        bint is_variable, no_min
         ndarray[int64_t] start, end
         ndarray[double_t] output
 
+    if minp == 0:
```

Comment:
@jreback thoughts on this approach? `no_min` is a boolean that indicates whether or not to consider `min_p` when choosing if the result is valid in `calc_sum`

Reply:
That's what I had earlier with the `minp2` stuff.

The problem there was that was that in `roll_sum`, `minp` gets set to `min(minp, 1)`, so we don't know if the user specified `min_periods=0`.

Reply:
hmm I don't see that. maybe can just remove that?

Reply:
@jreback, a25e7931ed26acf6b980674d777f822edfe72dc6 should be better.

Reply:
It looks bigger than it is because I had to ensure `VariableWindowIndexer` passes through `floor`. The main change is just using `floor=0` in `roll_sum` and iterating like `range(min(1, minp), ...)` instead of just `range(minmp, ...)`.